plugins/modules/ec2_instance.py

```diff
@@ -1289,6 +1299,15 @@ def build_top_level_options(params):
         spec['InstanceInitiatedShutdownBehavior'] = params.get('instance_initiated_shutdown_behavior')
     if params.get('termination_protection') is not None:
         spec['DisableApiTermination'] = params.get('termination_protection')
+    if params.get('hibernation_options') and params.get('volumes'):
+        for vol in params['volumes']:
+            if vol.get('ebs') and vol['ebs'].get('encrypted'):
+                spec['HibernationOptions'] = {'Configured': True}
+            else:
+                module.fail_json(
```

Comment:
Not sure if we want to fail here. Do we want to only warn the user that `hibernation_options` cannot be enabled because the volume is not encrypted and allow the user to perform the task ignoring `hibernation_options`? @tremble @jillr

Reply:
Usually we'd throw an error.  What would actually happen if they try to enable the option without encryption?

Reply:
hibernation_options : true without volume.ebs.encryptions (true) is not accepted by AWS. It throws an error and fails.